src/tvheadend/HTSPVFS.cpp

```diff
@@ -108,15 +114,56 @@ ssize_t HTSPVFS::Read ( unsigned char *buf, unsigned int len )
   if (read > 0)
     m_offset += read;
 
+  /* Tvheadend may briefly return 0 bytes when playing an in-progress recording at end-of-file
+     we'll retry 50 times with 10ms pauses (~500ms) before giving up */
+  if (read == 0 && inprogress)
+  {
+    for (int i = 1; i <= 50; i++)
+    {
+      read = SendFileRead(buf, len);
+      if (read > 0)
+      {
+        m_offset += read;
+        break;
+      }
+      usleep(10000);
+    }
+  }
```

Comment:
Maybe you should log a warning when leaving after 50 unsuccesful retries?

Reply:
Done... adjusted break to return, and added a log message after the loop